pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/ColumnIndexCreationInfo.java

```diff
@@ -37,11 +37,12 @@
   private final PartitionFunction partitionFunction;
   private final List<IntRange> partitionRanges;
   private final Object defaultNullValue;
+  private int numPartitions;
```

Comment:
Not sure if this is always number of Kafka partitions. If so, then can we just be explicit with numKafkaPartitions? if not, then maybe we can think of another name?